lib/components-types-v1_0_x.ts

```diff
@@ -83,36 +83,62 @@ export interface ComponentsDefinitionV10X {
           /**
            * Dropdown with fixed number of options
            */
-          type?: "select";
-          options?: any[];
+          type: "select";
+          options: {
+            /**
+             * Label of the item
+             */
+            caption: string;
+            /**
+             * Value of the item
+             */
+            value?: string;
+          }[];
         }
       | {
           /**
            * Checkbox toggling between value and no value
            */
-          type?: "checkbox";
-          value?: string;
+          type: "checkbox";
+          value: string;
         }
       | {
           /**
            * Text field property
            */
-          type?: "text";
-          [k: string]: any;
+          type: "text";
+          /**
+           * Value validation regexp pattern
+           */
+          pattern?: string;
+          /**
+           * Default value which is used instead of empty value
+           */
+          defaultValue?: string;
+          /**
+           * Unit type
+           */
+          unit?: string;
+          /**
+           * Input placeholder
+           */
+          inputPlaceholder?: string;
         }
       | {
           /**
            * Time field property
            */
-          type?: "time";
+          type: "time";
           [k: string]: any;
         }
       | {
           /**
            * Color picker field property
            */
-          type?: "colorPicker";
-          [k: string]: any;
+          type: "colorPicker";
+          options?: {
```

Comment:
Why is the opacity in an options object while for other fields properties are siblings?

Reply:
Guess we could drop the options thing?